### ISSUE TITLE
Intermittent Unchecked runtime.lastError: Could not establish connection. Receiving end does not exist.

### ISSUE THREAD

**Issue by aathan** [vu3th/vue-dapp#81]
Not clear whether this is a vue-dapp issue per-se but I intermittently receive this error which appears to be related to not checking window.chrome.runtime.lastError in the sendMessage() callback. To cause this problem I have to start with a fresh chrome window, call useBoard().open() and click the MetaMask provider (the only one I configure). There seems to be some kind of race condition somewhere, because the problem is intermittent. It seems to happen more frequently when the debug console is NOT already up.

**Comment by re2005** [comment]
@aathan please provide more details and a reproduction link where we can see the code.

**Comment by aathan** [comment]
I'm happy to help where I can. I believe there should be enough detail in my prior description to at least do a preliminary search in the vue-dapp codebase (or its dependencies) for sendMessage() and review those uses for checking of the return values.

Unfortunately I can't share proprietary code and it's not trivial for me to create a public extract of the use-case. I have limited time to try to build a separable reproduction of the issue but I will certainly try to find the time.

I ended up posting this report here "early" because it seemed like something you'd have insight on and be able to preliminarily investigate quickly. My own quick attempts to find use of sendMessage() came up empty, but it seems likely that communicating with MetaMask would involve the in-browser messaging APIs--but maybe everything is done via its injected ethers object? Maybe it's within MetaMask itself?

If there are instrumentations I can make and report back the results, that may be easier than creating an extractable/reproducible case. Let me know.
